dvc/scm/git/backend/dulwich/__init__.py

```diff
@@ -342,6 +343,8 @@ def get_ref(self, name, follow: bool = True) -> Optional[str]:
             except ValueError:
                 pass
         if ref:
+            if isinstance(self.repo[ref], Tag):
+                ref = self.repo[ref].object[1]
```

Comment:
Does this makes sense @pmrowla ?

Reply:
This is the right idea, but it's possible to have a longer chain of tag objects pointing to other tags which eventually point to a commit, so peeling the ref requires resolving the entire chain.

We should be able to do something like

```suggestion
            if isinstance(self.repo[ref], Tag):
                ref = self.repo.get_peeled(ref)
```

(note that I haven't tested this, you may need to dig a bit more into how `repo.get_peeled()` or `repo.refs.get_peeled()` works):
https://github.com/dulwich/dulwich/blob/ee231198600efcfa4874130f7fc619d8fee30c31/dulwich/repo.py#L708

Reply:
It works (at least for the test cases included)